### PR TITLE
Flow Matching, Diffusion PR

### PR DESCRIPTION
**Pull request by rpuns**
- added UNet Diffusion and Flow Matching
- added cross attention flow transformer
- cleaned up hard coded losses
- added and tested relevant configs

### CODE REVIEW

File: egomimic/algo/hpt.py [hunk 1]
```diff
@@ -116,7 +118,14 @@ def __init__(
         self.auxiliary_ac_keys = None
         self.shared_action = False
         self.device = None
-
+        
+        self.ot_6dof = False
+        self.use_dtw = False
```
Comment (ElmoPA) [inline egomimic/algo/hpt.py:123]: If this is for the ot then mayb make it a mode so that we can flexibly have it so that we can use different pairing methods like L2 and Language dist.

File: egomimic/algo/hpt.py [hunk 2]
```diff
@@ -271,7 +280,7 @@ def preprocess_tokens(self, domain, features):
 
         if self.token_postprocessing == "action_token":
             action_tokens = self.action_tokens.repeat(len(tokens), 1, 1)
-            tokens = torch.cat([tokens, action_tokens], dim=-2)
+            tokens = torch.cat([action_tokens, tokens], dim=-2)
```
Comment (ElmoPA) [inline egomimic/algo/hpt.py:283]: I will prefer to have action tokens later seems more like intuitively make sense (just feel that inputs is left and outputs is to the right).

File: egomimic/algo/hpt.py [hunk 3]
```diff
@@ -377,7 +386,23 @@ def stem_process(self, domain, data):
             feat_dict[modality] = stem_token
 
         return feats, feat_dict
+    
+    def resume_from_depth(self, block_outputs, depth):
+        """
+        Detach at trunk depth and resume trunk forward pass.
+        Gradients will only flow from depth upward.
+        """
+        cut_tokens = block_outputs[depth - 1].detach()
```
Comment (ElmoPA) [inline egomimic/algo/hpt.py:395]: What are we using stop gradient for? Also isn't it kinda inefficient to already compute the entire trunk but only use the one at that depth?

File: egomimic/algo/hpt.py (hunk 2 above)
Comment (rpuns) [reply egomimic/algo/hpt.py:283]: that’s a problem because then the action tokens for each embodiment don’t align - for example robot has additional wrist image which would mean by the aftention matrix, action tokens for human go in the same spot as wrist for robot

File: egomimic/algo/hpt.py (hunk 3 above)
Comment (rpuns) [reply egomimic/algo/hpt.py:395]: this is for the freeze at depth part for coat / ot loss

File: egomimic/algo/hpt.py (hunk 1 above)
Comment (rpuns) [reply egomimic/algo/hpt.py:123]: it is a mode but language pairing etc is not relevant for egoverse - will add for egobridge release
